### PR TITLE
Infinite News Item

### PR DESCRIPTION
**Pull request by avikumar15**

### CODE REVIEW

File: app/src/main/java/org/pragyan/dalal18/fragment/OrdersFragment.kt [hunk 1]
```diff
@@ -76,6 +90,8 @@ class OrdersFragment : Fragment(), SwipeRefreshLayout.OnRefreshListener, OrderIt
         model = activity?.run { ViewModelProvider(this).get(DalalViewModel::class.java) }
                 ?: throw Exception("Invalid activity")
 
+        recyclerView = rootView.findViewById(R.id.orders_recyclerView)
```
Comment (Harsh2098) [inline app/src/main/java/org/pragyan/dalal18/fragment/OrdersFragment.kt]: Is this needed?

File: app/src/main/java/org/pragyan/dalal18/fragment/OrdersFragment.kt [hunk 2]
```diff
@@ -76,6 +90,8 @@ class OrdersFragment : Fragment(), SwipeRefreshLayout.OnRefreshListener, OrderIt
...
+        frameLayoutOpenOrders = rootView.findViewById(R.id.flOrdersFrag)
```
Comment (Harsh2098) [inline app/src/main/java/org/pragyan/dalal18/fragment/OrdersFragment.kt]: Shift this to `onViewCreated()` ?

File: app/src/main/java/org/pragyan/dalal18/fragment/OrdersFragment.kt [hunk 3]
```diff
@@ -196,8 +216,17 @@ class OrdersFragment : Fragment(), SwipeRefreshLayout.OnRefreshListener, OrderIt
...
 
+        if (preferences.getBoolean(Constants.SWIPE_TO_DELETE_ORDER, true)) {
+            preferences.edit().putBoolean(Constants.SWIPE_TO_DELETE_ORDER, false).apply()
+            deleteTutorial = DeleteTutorial(con, this)
+            frameLayoutOpenOrders.addView(deleteTutorial)
+        }
+    }
+
+    override fun onSwiped(viewHolder: RecyclerView.ViewHolder, direction: Int, position: Int) {
+// TODO
```
Comment (Harsh2098) [inline app/src/main/java/org/pragyan/dalal18/fragment/OrdersFragment.kt]: TODO ?

File: app/src/main/java/org/pragyan/dalal18/fragment/HomeFragment.kt [hunk 4]
```diff
@@ -237,7 +237,7 @@ class HomeFragment : Fragment(), NewsRecyclerAdapter.NewsItemClickListener, Swip
 
     override fun onCompanyTickerClick(view: View, position: Int) {
         val bundle = Bundle()
-        bundle.putString(CompanyDescriptionFragment.COMPANY_NAME_KEY, companyTickerDetailsList[position].fullName)
+        bundle.putString(CompanyDescriptionFragment.COMPANY_NAME_KEY, companyTickerDetailsList[position%companyTickerDetailsList.size].fullName)
```
Comment (Harsh2098) [inline app/src/main/java/org/pragyan/dalal18/fragment/HomeFragment.kt]: Reformat code before pushing

File: app/src/main/java/org/pragyan/dalal18/adapter/OrdersRecyclerAdapter.java [hunk 5]
```diff
@@ -125,6 +125,7 @@ public boolean updateOrder(MyOrderUpdate order, String companyName) {
                 openOrdersList.remove(position);
                 notifyItemRemoved(position);
             } else if (position != -1) {
+                System.out.println("the stock qty to add is "+order.getStockQuantity());
```
Comment (Harsh2098) [inline app/src/main/java/org/pragyan/dalal18/adapter/OrdersRecyclerAdapter.java]: Remove this if not needed

**Comment by Harsh2098** [comment]
```
for (x in 1..3)
      companyTickerDetailsList.addAll(companyTickerDetailsList)
```

Also, remove this if new logic doesn't require this

File: app/src/main/java/org/pragyan/dalal18/fragment/OrdersFragment.kt (hunk 1 above)
Comment (avikumar15) [reply app/src/main/java/org/pragyan/dalal18/fragment/OrdersFragment.kt]: removed!

File: app/src/main/java/org/pragyan/dalal18/fragment/OrdersFragment.kt (hunk 3 above)
Comment (avikumar15) [reply app/src/main/java/org/pragyan/dalal18/fragment/OrdersFragment.kt]: removed lol

File: app/src/main/java/org/pragyan/dalal18/fragment/OrdersFragment.kt [hunk 6]
```diff
@@ -58,8 +66,13 @@ class OrdersFragment : Fragment(), SwipeRefreshLayout.OnRefreshListener, OrderIt
     private var networkDownHandler: ConnectionUtils.OnNetworkDownHandler? = null
     private lateinit var loadingOrdersDialog: AlertDialog
 
+    private lateinit var con: Context
+    private lateinit var frameLayoutOpenOrders: FrameLayout
+    private lateinit var deleteTutorial: DeleteTutorial
+
     override fun onAttach(context: Context) {
         super.onAttach(context)
+        con=context
```
Comment (Harsh2098) [inline app/src/main/java/org/pragyan/dalal18/fragment/OrdersFragment.kt]: Can we do away with this?

File: app/src/main/java/org/pragyan/dalal18/fragment/OrdersFragment.kt [hunk 7]
```diff
@@ -230,6 +242,13 @@ class OrdersFragment : Fragment(), SwipeRefreshLayout.OnRefreshListener, OrderIt
         }
     }
 
+    private fun showSwipeToDeleteTour() {
```
Comment (Harsh2098) [inline app/src/main/java/org/pragyan/dalal18/fragment/OrdersFragment.kt]: Did you rebase wrt to master? Why is swipeToDeleteTour() here?

Comment (avikumar15) [reply app/src/main/java/org/pragyan/dalal18/fragment/OrdersFragment.kt]: Yes. I didn't call the function just didn't use it

**Pull request closed without merging**